### PR TITLE
Add SpanEvent serializer in benchmark module

### PR DESCRIPTION
**Pull request by ambushwork**
### What does this PR do?

Add SpanEvent serializer in benchmark module

### Additional Notes

* `DatadogDataConstraints` constructor is assigned with default value of `internalLogger`, so that in benchmark module, it doesn't have to provide an implementation of `InternalLogger`

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt [hunk 1]
```diff
@@ -16,7 +17,9 @@ import java.util.Locale
  *
  * @param internalLogger Internal logger.
  */
-class DatadogDataConstraints(private val internalLogger: InternalLogger) : DataConstraints {
+class DatadogDataConstraints(
+    private val internalLogger: InternalLogger = NoOpInternalLogger()
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt:21]: maybe it is better to revert this change and just explicitly provide `NoOpInternalLogger` at the call site in the `benchmark` module?

File: dd-sdk-android-internal/api/dd-sdk-android-internal.api
Comment (0xnm) [inline dd-sdk-android-internal/api/dd-sdk-android-internal.api:1]: empty file in the commit?

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt (hunk 1 above)
Comment (ambushwork) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt:21]: `InernalLogger` and `NoOpInternalLogger` are now inside `core`, `internal` doesn't have dependency on it, we need to migrate them first if we want to do that, what do you prefer?

Comment (ambushwork) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt:21]: Yes, sorry, I was confused by myself, ignore what I said above. it's actually the problem of `internal` modifier on `NoOpInternalLogger`

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.Date
+
+internal class SpanEventSerializer(
+    private val dataConstraints: DataConstraints = DatadogDataConstraints()
+) {
+
+    // region Serializer
+
+    fun serialize(env: String, spanEvents: List<SpanEvent>): String {
+        val spans = JsonArray(1)
```
Comment (mariusc83) [inline tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt:25]: why `JsonArray(1)` here ?

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        jsonObject.addProperty(TAG_ENV, env)
+
+        return jsonObject.toString()
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun sanitizeKeys(model: SpanEvent): SpanEvent {
```
Comment (mariusc83) [inline tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt:40]: let's name this `span` instead of `model`

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        val newUserObject = sanitizeUserAttributes(model.meta.usr)
+        val newMetricsObject = sanitizeMetrics(model.metrics)
+        return model.copy(meta = model.meta.copy(usr = newUserObject), metrics = newMetricsObject)
+    }
+
+    private fun sanitizeUserAttributes(usr: SpanEvent.Usr): SpanEvent.Usr {
```
Comment (mariusc83) [inline tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt:46]: do you really need to sanitize here ? These are internal spans so I am not sure we need this. WDYT ?

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt (hunk 2 above)
Comment (ambushwork) [reply tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt:25]: It seems we only send one span in each `spans` attribute, refer to :
https://github.com/DataDog/dd-sdk-android/blob/develop/features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/internal/domain/event/SpanEventSerializer.kt

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt (hunk 4 above)
Comment (ambushwork) [reply tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt:46]: I am not sure, I copy them from `feature-traces` in case that we might need it, I can remove it since it's not useful

File: tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt (hunk 2 above)
Comment (ambushwork) [reply tools/benchmark/src/main/java/com/datadog/benchmark/internal/SpanEventSerializer.kt:25]: you are right, now changed to `spanEvents.size`

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt (hunk 1 above)
Comment (ambushwork) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/constraints/DatadogDataConstraints.kt]: this is removed since Marius said sanitize is not necessary here.

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Review by mariusc83 — APPROVED** [review]
